reactor-netty-http/src/test/java/reactor/netty/http/client/HttpClientTest.java

```diff
@@ -2489,4 +2493,89 @@ public void testLoopAndResolver() {
 		loop.disposeLater()
 		    .block(Duration.ofSeconds(30));
 	}
+
+	@Test
+	public void testCustomMetricsWithUriMapper() {
+		disposableServer =
+				HttpServer.create()
+							.port(0)
+							.handle((req, resp) -> resp.sendString(Mono.just("OK")))
+							.wiretap(true)
+							.bindNow();
+
+		Sinks.Many<String> collectedUris = Sinks.many().replay().all();
+
+		HttpClient.create()
+				.metrics(true,
+						() -> new HttpClientMetricsRecorder() {
+							@Override
+							public void recordDataReceived(SocketAddress remoteAddress, String uri, long bytes) {
+								collectedUris.emitNext(uri);
+							}
+
+							@Override
+							public void recordDataSent(SocketAddress remoteAddress, String uri, long bytes) {
+								collectedUris.emitNext(uri);
+							}
+
+							@Override
+							public void incrementErrorsCount(SocketAddress remoteAddress, String uri) {
+								collectedUris.emitNext(uri);
+							}
+
+							@Override
+							public void recordDataReceived(SocketAddress remoteAddress, long bytes) {
```

Comment:
@violetagg do we have an abstract/no-op base class (either in prod or test sources) that could remove the need to implement all these empty methods?

Reply:
no there isn't